File: partie-1/models/inscription.py
from sqlalchemy import Integer, Column, Text
from sqlalchemy.orm import Mapped, mapped_column

from db import Base


class Inscription(Base):
    __tablename__ = "liste_inscriptions"

    annee_etude: Mapped[int] = mapped_column(Integer, primary_key=True)
    matricule: Mapped[str] = mapped_column(Text, primary_key=True)
    cours_json: Mapped[str] = mapped_column(Text, nullable=False)
    nom: Mapped[str] = mapped_column(Text, nullable=False)
    prenom: Mapped[str] = mapped_column(Text, nullable=False)

File: partie-3/app/schemas/inscription.ts
import {z} from 'zod'

export const InscriptionSchema = z.object({
  matricule: z.string(),
  nom: z.string(),
  prenom: z.string(),
  annee_etude: z.number(),
  cours_json: z.string()
})

export type Inscription = z.infer<typeof InscriptionSchema>

export const InscriptionArraySchema = z.array(InscriptionSchema)